src/core/execute.c

```diff
@@ -1428,6 +1428,57 @@ static int apply_protect_sysctl(Unit *u, const ExecContext *c) {
         return r;
 }
 
+static int apply_protect_kernel_modules(Unit *u, const ExecContext *c) {
+        const SystemCallFilterSet *set;
+        scmp_filter_ctx *seccomp;
+        const char *sys;
+        bool syscalls_found = false;
+        int r;
+
+        assert(c);
+
+        if (skip_seccomp_unavailable(u, "ProtectKernelModules="))
+                return 0;
+
+        seccomp = seccomp_init(SCMP_ACT_ALLOW);
+        if (!seccomp)
+                return -ENOMEM;
+
+        r = seccomp_add_secondary_archs(seccomp);
+        if (r < 0)
+                goto finish;
+
+        for (set = syscall_filter_sets; set->set_name; set++)
+                if (streq(set->set_name, "@module")) {
+                        syscalls_found = true;
+                        break;
+                }
```

Comment:
hmm, I don't think it is worth relying on the syscall_filter_sets structure for this. I'd probably just hardcode the three syscalls here, it's not that many, and it's very unlikely it will change anytime soon. Also, the linear search goes away then.